githooks/prepare-commit-msg

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+#
+# A hook that adds a commit template.
+
+COMMIT_MSG_FILE=$1
+COMMIT_SOURCE=$2
+SHA1=$3
+
+# Remove the default message.
+/usr/bin/perl -i.bak -ne 'print unless(m/^. Please enter the commit message/..m/^#$/)' "$COMMIT_MSG_FILE"
+
+# Print our custom commit template.
+/usr/bin/perl -le 'open FH, "./commit-template.txt"; while(<FH>){print "foo"}; print "\n"'
```

Comment:
What happens if this is unavailable for whatever reason on some distros? Does it prevent the dev from commiting?

Reply:
Yeah, it would. If the script exits with an error code git won't let you commit. I'm sure there's a flag to override it.

But if the user installs a script that just doesn't work on his system, then he needs to uninstall the hook. (Maybe we should have a just command for this too. Unsure.)

Reply:
`git commit -n` overrides it 

```
-n, --no-verify       bypass pre-commit and commit-msg hooks
```